R/demographics-table.R

```diff
@@ -0,0 +1,326 @@
+#' Check if summary function evaluates to the correct format
+#' @param fun The summary function to use for summarizing the continuous data in [pt_demographics()]
+#' @keywords internal
+validate_dem_fun <- function(fun){
+  see <- "See ?pmtables:::dem_cont_fun."
+  result <- fun(seq(1,5))
+  assert_that(
+    !is_empty(result),
+    msg = glue("`fun` returned an empty object. {see}")
+  )
+  assert_that(
+    is.data.frame(result),
+    msg = glue("`fun` must return a data frame. {see}")
+  )
+  assert_that(
+    nrow(result)==1,
+    msg = glue("`fun` must return a data frame with a single row. {see}")
+  )
+}
+
+#' Default summary function in pt_demographics for continuous variables
+#'
+#' @param value a vector or column to summarize
+#' @param digits the number of digits in the summary; the current implementation
+#' passes digits to [sig()]
+#'
+#' @return
+#' A tibble with one row and three columns:
+#'  1. `mean (sd)` the mean and standard deviation of `value`
+#'  2. `min-max` the range of the `value`
+#'  3. `non-missing` the number of non-missing `values`
+#'
+#' @details
+#' The summary function must return a data frame (or tibble) with a single
+#' row and data summaries in the columns. Results will be coerced to
+#' character.
+#'
+#' @examples
+#' pmtables:::dem_cont_fun(value = seq(1,7), digits = 2)
+#'
+#' @keywords internal
+dem_cont_fun <- function(value = seq(1,5), digits = 3) {
+  Mean  <- sig(mean(value,  na.rm = TRUE), digits = digits)
+  Sd    <- sig(sd(value,    na.rm = TRUE), digits = digits)
+  Range <- sig(range(value, na.rm = TRUE), digits = digits)
+  tibble(
+    `mean (sd)` = paste0(Mean, " (", Sd, ")"),
+    `min-max` = paste0(Range, collapse = " - "),
```

Comment:
Per the style guide, could these both be changed to en-dashes to indicate range, to distinguish from hyphen?
```
    `min--max` = paste0(Range, collapse = " -- "),
```

Reply:
The current pmtables functions actually show `Min / Max` should we use that in this function too so all tables are consistent? 
![Screen Shot 2021-08-09 at 9 57 24 AM](https://user-images.githubusercontent.com/42553148/128718208-87c82b24-5381-435f-886e-002e911a35f5.png)

Is it possible to make all of these outputs consistent with the other tables? So `Mean (SD)`, `Min / Max` and `Non-missing`

Reply:
Two more thoughts on the non-missing row

1. Grammatically it should read `Nonmissing` (without the hyphen)
2. Is it too late to swap this to show `Missing` as we do in other tables? So in this table, instead of 65, 92 and 157 nonmissing in the Weight row, could it read 1, 2, 3 Missing? 

![Screen Shot 2021-08-09 at 10 21 00 AM](https://user-images.githubusercontent.com/42553148/128722246-38151518-e6ce-482a-8324-76fe7bd65e2c.png)

Reply:
Will change to `Min / Max` per @KatherineKayMRG suggestion.

Reply:
Also will report `Missing` per @KatherineKayMRG suggestion.